File: src/main/java/com/bookstore/booksmanagementsystem/service/SchoolService.java
package com.bookstore.booksmanagementsystem.service;

import com.bookstore.booksmanagementsystem.dto.SchoolDTO;
import java.util.List;

public interface SchoolService {
	List<SchoolDTO> getAllSchools();

	SchoolDTO getSchoolByName(String name);

	SchoolDTO getSchoolById(Long id);

	List<SchoolDTO> searchSchoolsByName(String partialName);

	SchoolDTO createSchool(SchoolDTO schoolDTO);

	SchoolDTO updateSchool(Long id, SchoolDTO schoolDTO);

	void deleteSchool(Long id);
}

File: bookstore-frontend/src/services/schoolService.js
import api from './api';

const schoolService = {
  getAllSchools: () => api.get('/schools'),
  getSchoolById: (id) => api.get(`/schools/${id}`),
  createSchool: (schoolData) => api.post('/schools', schoolData),
  updateSchool: (id, schoolData) => api.put(`/schools/${id}`, schoolData),
  deleteSchool: (id) => api.delete(`/schools/${id}`),
};

export default schoolService;
